src/Services/Cda/CdaTemplateParse.php

```diff
@@ -373,12 +373,14 @@ public function fetchEncounterPerformed($entry): void
                 $i += count($this->templateData['field_name_value_array']['encounter']);
             }
 
-            $this->templateData['field_name_value_array']['encounter'][$i]['extension'] = $entry['encounter']['id']['extension'];
-            $this->templateData['field_name_value_array']['encounter'][$i]['root'] = $entry['encounter']['id']['root'];
-            $this->templateData['field_name_value_array']['encounter'][$i]['date'] = ($entry['encounter']['effectiveTime']['value'] ?? null) ?: $entry['encounter']['effectiveTime']['low']['value'] ?? null;
-            $this->templateData['field_name_value_array']['encounter'][$i]['date_end'] = $entry['encounter']['effectiveTime']['high']['value'] ?? null;
+            if (!empty($this->templateData['field_name_value_array']['encounter'][$i])) {
+                $this->templateData['field_name_value_array']['encounter'][$i]['extension'] = $entry['encounter']['id']['extension'];
+                $this->templateData['field_name_value_array']['encounter'][$i]['root'] = $entry['encounter']['id']['root'];
+                $this->templateData['field_name_value_array']['encounter'][$i]['date'] = ($entry['encounter']['effectiveTime']['value'] ?? null) ?: $entry['encounter']['effectiveTime']['low']['value'] ?? null;
+                $this->templateData['field_name_value_array']['encounter'][$i]['date_end'] = $entry['encounter']['effectiveTime']['high']['value'] ?? null;
+            }
```

Comment:
Can't do.  This may/will initially be empty but is init'ed by import data.

This is the parser for the import and the $this->templateData is unknowable and we should allow all arrays to be populated even with empty or null because those values may be a trigger in the actual DB import which happens after user reviews.

Please remove all conditionals put in and allow those arrays be init'ed. If they aren't they may end up causing warnings in DB import if they are expected but not there.

I hope I've been clear and if not please let me know.

Reply:
Looks like this is only place that concerns me otherwise thanks for fixing these.

Reply:
Just want to show how this break encounters. Otherwise import to patient is working better than I hoped but I see on area I need to investigate, vitals only populates using first encounter and not all vitals which I do parse and show in ccda.
I may do some fixes and push to your branch if that's okay?

![image](https://github.com/openemr/openemr/assets/1859985/8ad1a6d8-d1c7-4519-b61f-c01097eea8eb)

Reply:
You'd think Stephen I'd have better things to do besides picking on!:)